### ISSUE TITLE
SqlServer: Scaffolding default schema

### ISSUE THREAD

**Issue by smitpatel** [aspnet/EntityFramework#9318]
At present we identify the default schema using `SELECT SCHEMA_NAME()`
But default schema is user specific. So different users may have different schema. This can give rise to different scaffolded code based on who ran the rev eng & cause ugly version control diffs especially once update model for database is implemented.

If we stop scaffolding default schema then if the schema is anything other than "dbo" (which is our default) then it will be used in scaffolding objects. (like `ToTable("tablename", "nondefaultschema")`) but it will produce more consistent code across users.

Another option is to take command line argument for default schema so user can specify what to consider as default schema as opposed to we trying to guess based on user. But we can defer this to customer feedback.

@bricelam - comments?

**Comment by bricelam** [comment]
*dbo* is not our default (except for functions). If you don't specify a schema, we don't specify one in SQL and the default schema of the current user is used.

The way we currently scaffold means that no matter what user you use to reverse engineer, it should work for all users if they have access to the objects.

It may be helpful to have an option never to use the default schema (always configure it on each table).

**Comment by ErikEJ** [comment]
I think the schema should always be scaffolded.

**Comment by divega** [comment]
@ErikEJ the schema on each table or the default one? Can you elaborate a bit more on why? :smile:

**Comment by ErikEJ** [comment]
I meant be explicit on each table... Because this is scaffolding, and should reflect the actual layout of the db.

**Comment by divega** [comment]
@ErikEJ I don't think that is necessarily bad, but we also as a principle try to avoid generating unnecessary repetitive code when the default conventions makes it unnecessary.

**Comment by divega** [comment]
> If you don't specify a schema, we don't specify one in SQL

FWIW, in SQL Server there is supposedly a performance overhead when schema is not specified in the SQL, because it needs to run some heuristics to resolve the table. I can't remember if ever tried to measure that.

**Comment by divega** [comment]
@bricelam will revisit this when he is working on update model from database.
